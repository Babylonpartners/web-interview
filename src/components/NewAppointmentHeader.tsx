import * as React from 'react'
import { headerStyles } from '../styles/styles'
import { Responsive } from 'semantic-ui-react';

interface Props {
    user: any
}


export class NewAppointmentHeader extends React.PureComponent<Props> {
    render() {
        return (
        <>
        
        
        <br />  
        <br /> 
        <Responsive maxWidth={640}>
        <h2 
            data-testid="newAppointmentHeader"
            className="mobile-new-appointment-header"
            style={headerStyles}
        >
            New appointment
        </h2>
       {this.props.user &&
        <div 
            data-testid="userPanel"
            className="user-panel"
        >
          <img  
            data-testid="userAvatar"
            src={this.props.user.avatar} 
            className="user-image" 
            alt="user" 
          />
          <h2
            data-testid="userName" 
            style={headerStyles} 
            className="user-name"
          >
          {this.props.user.firstName} {this.props.user.lastName}
          </h2>
        </div>
        }
          <br />
        <hr 
           style={headerStyles} 
        />
        </Responsive>


        <Responsive minWidth={641}>
        <h2 
            data-testid="newAppointmentHeader"
            className="new-appointment-header"
            style={headerStyles}
        >
            New appointment
        </h2>
        
        <div 
            data-testid="userPanel"
            className="user-panel"
        >
          <img 
            data-testid="userAvatar"
            src={this.props.user.avatar} 
            className="user-image" 
            alt="user" 
         />
          <h2 
            data-testid="userName"
            style={headerStyles}
            className="user-name"
          >
          {this.props.user.firstName} {this.props.user.lastName}
          </h2>
        </div>
        
        <br />
        <hr 
           style={headerStyles} 
        />
        </Responsive>


        </>
        
        );
}}