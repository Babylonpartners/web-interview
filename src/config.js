export const API_ENDPOINT = 'https://web-interview-server.herokuapp.com'
export const API_ENDPOINT_DEV = 'http://localhost:3001'
