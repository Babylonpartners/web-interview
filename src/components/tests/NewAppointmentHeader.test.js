import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { NewAppointmentHeader } from '../NewAppointmentHeader'
import { API_ENDPOINT } from '../../config'

afterEach(cleanup)

const user = {
  id: 1,
  firstName: 'Jane',
  lastName: 'Doe',
  dateOfBirth: '1971-02-03',
}

test('<NewAppointmentHeader>', () => {
  const { queryByTestId, getByTestId, container } = render(
    <NewAppointmentHeader user userLoading />
  )
  expect(container.firstChild).toMatchSnapshot()

  expect(getByTestId('newAppointmentHeader')).toBeTruthy()
  expect(getByTestId('newAppointmentHeader').textContent).toBe(
    'New appointment'
  )

  expect(queryByTestId('userPanel')).toBeTruthy()

  expect(queryByTestId('userAvatar')).toBeTruthy()
  expect(user.dateOfBirth).toEqual('1971-02-03')
  expect(queryByTestId('userName')).toBeTruthy()
  expect(user.firstName).toEqual('Jane')
})
