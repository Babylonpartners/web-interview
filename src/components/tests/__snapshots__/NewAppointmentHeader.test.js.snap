// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewAppointmentHeader> 1`] = `<br />`;
